pkg/combustion/kubernetes.go

```diff
@@ -88,45 +98,146 @@ func configureRKE2(ctx *image.Context) (string, error) {
 		return "", fmt.Errorf("copying RKE2 installer script: %w", err)
 	}
 
-	config, err := parseKubernetesConfig(ctx)
-	if err != nil {
-		return "", fmt.Errorf("parsing RKE2 config: %w", err)
+	if len(ctx.ImageDefinition.Kubernetes.Nodes) > 1 {
+		return configureMultiNodeRKE2(ctx)
```

Comment:
Not something we have to solve for now, but if this was 2 this code path would still execute, whereas it wouldn't be advised to implement two-nodes without NATS integration, for example. No big deal, but maybe a warning would be good about two-nodes with etcd not actually giving you any form of HA (as if one etcd node goes down, the other will too, as you lose quorum).

Reply:
Do we actively discourage an architecture of 1 server and 1 agent? It has nothing to do with a HA as usually defined but it's a topology that users might want to use for some reason. Thoughts?

Reply:
Oh! I was just thinking of multi-server nodes. You're right to think about one server and one agent use-cases, this is actually something we know that customers would be interested in. Where they need the capacity but not necessarily availability of the control plane. I think the check would be for two *servers*.

Reply:
Makes sense but this is something we should do during the validation step. I'll address it.